libcaf_net/caf/net/lp/client_factory.cpp

```diff
@@ -33,6 +33,13 @@ class client_factory::config_impl : public dsl::client_config_value {
   using super = dsl::client_config_value;
 
   using super::super;
+
+  explicit config_impl(multiplexer* mpx, size_t max_message_length)
```

Comment:
Rather than configuring the length, the ticket was about selecting the integer type on the wire, i.e., 1, 2, 4 or 8 byte.

This also shouldn't be an integer but an enum type in order to make it clear what options are available to the user, e.g.:

```c++
/// Configures the integer type used as the length prefix.
enum class size_field_type {
  /// Configures a length prefix with a single byte.
  u1,
  // Configures a length prefix with two bytes.
  u2,
  // Configures a length prefix with four bytes.
  u4,
  // Configures a length prefix with eight bytes.
  u8,
};
```